common/types.ts

```diff
@@ -0,0 +1,16 @@
+/**
+ * A mapping between libraries and the addresses to which they were deployed.
+ *
+ * Containing support for two level configuration.
+ */
+export interface LibraryAddresses {
+  [libraryName: string]: string | { [location: string]: string };
+}
+
+/**
+ * A mapping between libraries and lists of placeholder instances present in their hex-encoded bytecode.
+ * For each placeholder its length and the position of the first character is stored.
+ */
+export interface LinkReferences {
```

Comment:
Some of the info I added in https://github.com/ethereum/solc-js/pull/603#discussion_r809277437 would probably be useful here. E.g. that `start` and `length` are actually positions in binary bytecode.

Reply:
This is marked as resolved but I don't think that much changed here. One very important piece of information from that docstring that definitely should be here is that `start` and `length` always refer to positions in binary and not hex-encoded bytecode.

Reply:
I have updated to include this.